### ISSUE TITLE
NoSuchMethodError: The getter 'isClosed' was called on null

### ISSUE THREAD

**Issue by zsirfs** [flutter/flutter#22253]
## Steps to Reproduce

1. flutter create project
2. flutter run 

## Logs

```
[14:07:45 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +6 ms] [DEVICE LOG] Timestamp                       (process)[PID]
[14:07:45 GMT+0800 (CST)] [FlutterRun] [Info] <== [ +223 ms] Skipping kernel compilation. Fingerprint match.
[14:07:46 GMT+0800 (CST)] [FlutterRun] [Info] <== [ +346 ms] Building bundle
[14:07:46 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +1 ms] Writing asset files to build/flutter_assets
[14:07:46 GMT+0800 (CST)] [FlutterRun] [Info] <== [ +108 ms] Wrote build/flutter_assets
[14:07:46 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +3 ms] executing: /usr/bin/xcrun simctl get_app_container 6DBBD95F-4D36-47CF-892C-171BFDE35AB4 com.luoyangfu.flutterApp
[14:07:46 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +2 ms] executing: /usr/bin/killall Runner
[14:07:48 GMT+0800 (CST)] [FlutterRun] [Info] <== [+1751 ms] executing: /usr/bin/xcrun simctl launch 6DBBD95F-4D36-47CF-892C-171BFDE35AB4 com.luoyangfu.flutterApp --enable-dart-profiling --enable-checked-mode --start-paused --observatory-port=0
[14:07:48 GMT+0800 (CST)] [FlutterRun] [Info] <== [ +385 ms] com.luoyangfu.flutterApp: 59451
[14:07:48 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +2 ms] Waiting for observatory port to be available...
[14:07:49 GMT+0800 (CST)] [FlutterRun] [Info] <== [+1191 ms] [DEVICE LOG] 2018-09-25 14:07:49.697720+0800  localhost Runner[59451]: (QuartzCore) [com.apple.coreanimation:WindowServer] Initialized 2 displays
[14:07:49 GMT+0800 (CST)] [FlutterRun] [Info] <== [  +44 ms] [DEVICE LOG] 2018-09-25 14:07:49.721247+0800  localhost Runner[59451]: (libAccessibility.dylib) [com.apple.Accessibility:AccessibilitySupport] Retrieving resting unlock: 0
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [ +501 ms] [DEVICE LOG] 2018-09-25 14:07:50.239362+0800  localhost Runner[59451]: (Runner) Created Activity ID: 0xf7600, Description: Updating Key-Value Observers Of Preferences
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [  +11 ms] [DEVICE LOG] 2018-09-25 14:07:50.239673+0800  localhost Runner[59451]: (Runner) Created Activity ID: 0xf7601, Description: Updating Key-Value Observers Of Preferences
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +8 ms] [DEVICE LOG] 2018-09-25 14:07:50.239835+0800  localhost Runner[59451]: (Runner) Created Activity ID: 0xf7602, Description: Updating Key-Value Observers Of Preferences
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +8 ms] [DEVICE LOG] 2018-09-25 14:07:50.240046+0800  localhost Runner[59451]: (Runner) Created Activity ID: 0xf7603, Description: Updating Key-Value Observers Of Preferences
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [  +10 ms] [DEVICE LOG] 2018-09-25 14:07:50.240196+0800  localhost Runner[59451]: (Runner) Created Activity ID: 0xf7604, Description: Updating Key-Value Observers Of Preferences
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [  +14 ms] [DEVICE LOG] 2018-09-25 14:07:50.240477+0800  localhost Runner[59451]: (Runner) Created Activity ID: 0xf7605, Description: Updating Key-Value Observers Of Preferences
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [  +16 ms] [DEVICE LOG] 2018-09-25 14:07:50.240708+0800  localhost Runner[59451]: (Runner) Created Activity ID: 0xf7606, Description: Updating Key-Value Observers Of Preferences
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [ +208 ms] [DEVICE LOG] 2018-09-25 14:07:50.516335+0800  localhost Runner[59451]: (FrontBoardServices) [com.apple.BackBoard:Display] [FBDisplayManager=0x6000038185c0] silently connecting <FBSDisplayConfiguration: 0x600001625450; Main; mode: "375x812@3x 60Hz p3 SDR">
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [ +114 ms] [DEVICE LOG] 2018-09-25 14:07:50.523354+0800  localhost Runner[59451]: (AXMediaUtilities) [com.apple.Accessibility:AXMediaCommon] Unable to look up screen scale
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +4 ms] [DEVICE LOG] 2018-09-25 14:07:50.523427+0800  localhost Runner[59451]: (AXMediaUtilities) [com.apple.Accessibility:AXMediaCommon] Unexpected physical screen orientation
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [  +30 ms] [DEVICE LOG] 2018-09-25 14:07:50.558109+0800  localhost Runner[59451]: (AXMediaUtilities) [com.apple.Accessibility:AXMediaCommon] Unable to look up screen scale
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [  +52 ms] [DEVICE LOG] 2018-09-25 14:07:50.564512+0800  localhost Runner[59451]: (AXMediaUtilities) [com.apple.Accessibility:AXMediaCommon] Unable to look up screen scale
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [  +10 ms] [DEVICE LOG] 2018-09-25 14:07:50.564556+0800  localhost Runner[59451]: (AXMediaUtilities) [com.apple.Accessibility:AXMediaCommon] Unexpected physical screen orientation
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [ +102 ms] [DEVICE LOG] 2018-09-25 14:07:50.799866+0800  localhost Runner[59451]: (UIKitCore) You've implemented -[<UIApplicationDelegate> application:performFetchWithCompletionHandler:], but you still need to add "fetch" to the list of your supported UIBackgroundModes in your Info.plist.
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +2 ms] [DEVICE LOG] 2018-09-25 14:07:50.800062+0800  localhost Runner[59451]: (UIKitCore) You've implemented -[<UIApplicationDelegate> application:didReceiveRemoteNotification:fetchCompletionHandler:], but you still need to add "remote-notification" to the list of your supported UIBackgroundModes in your Info.plist.
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [  +94 ms] [DEVICE LOG] 2018-09-25 14:07:50.935138+0800  localhost Runner[59451]: (Flutter) flutter: Observatory listening on http://127.0.0.1:60431/
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +6 ms] Observatory URL on device: http://127.0.0.1:60431/
[14:07:50 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +7 ms] Connecting to service protocol: http://127.0.0.1:60431/
[14:07:51 GMT+0800 (CST)] [FlutterRun] [Info] <== [ +453 ms] [DEVICE LOG] 2018-09-25 14:07:51.324634+0800  localhost Runner[59451]: (AccessibilityUtilities) [com.apple.Accessibility:AXLoading] Will reevaluate code items and load any needed ax code items now
[14:07:51 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +4 ms] [DEVICE LOG] 2018-09-25 14:07:51.325355+0800  localhost Runner[59451]: (AccessibilityUtilities) [com.apple.Accessibility:AXLoading] found 14 axbundle(s) requiring load
[14:07:51 GMT+0800 (CST)] [FlutterRun] [Info] <== [  +14 ms] [DEVICE LOG] 2018-09-25 14:07:51.418895+0800  localhost Runner[59451]: (AccessibilityUtilities) [com.apple.Accessibility:AXLoading] Finished loading ax code items
[14:07:51 GMT+0800 (CST)] [FlutterRun] [Info] <== [ +297 ms] [DEVICE LOG] 2018-09-25 14:07:51.717732+0800  localhost Runner[59451]: (AccessibilityUtilities) [com.apple.Accessibility:AXLoading] Will reevaluate code items and load any needed ax code items now
[14:07:51 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +1 ms] [DEVICE LOG] 2018-09-25 14:07:51.717914+0800  localhost Runner[59451]: (AccessibilityUtilities) [com.apple.Accessibility:AXLoading] found 0 axbundle(s) requiring load
[14:07:51 GMT+0800 (CST)] [FlutterRun] [Info] <== [   +2 ms] [DEVICE LOG] 2018-09-25 14:07:51.717965+0800  localhost Runner[59451]: (AccessibilityUtilities) [com.apple.Accessibility:AXLoading] Finished loading ax code items
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] ==> [{"id":"1","method":"app.callServiceExtension","params":{"appId":"10ad1be2-f8e0-4372-a286-607579a26a1f","methodName":"ext.flutter.debugPaint","params":{"enabled":false}}}]
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] ==> [{"id":"2","method":"app.callServiceExtension","params":{"appId":"10ad1be2-f8e0-4372-a286-607579a26a1f","methodName":"ext.flutter.showPerformanceOverlay","params":{"enabled":false}}}]
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] ==> [{"id":"3","method":"app.callServiceExtension","params":{"appId":"10ad1be2-f8e0-4372-a286-607579a26a1f","methodName":"ext.flutter.repaintRainbow","params":{"enabled":false}}}]
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] ==> [{"id":"4","method":"app.callServiceExtension","params":{"appId":"10ad1be2-f8e0-4372-a286-607579a26a1f","methodName":"ext.flutter.timeDilation","params":{"timeDilation":1}}}]
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] ==> [{"id":"5","method":"app.callServiceExtension","params":{"appId":"10ad1be2-f8e0-4372-a286-607579a26a1f","methodName":"ext.flutter.debugAllowBanner","params":{"enabled":true}}}]
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] ==> [{"id":"6","method":"app.callServiceExtension","params":{"appId":"10ad1be2-f8e0-4372-a286-607579a26a1f","methodName":"ext.flutter.debugPaintBaselinesEnabled","params":{"enabled":false}}}]
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] ==> [{"id":"7","method":"app.callServiceExtension","params":{"appId":"10ad1be2-f8e0-4372-a286-607579a26a1f","methodName":"ext.flutter.debugWidgetInspector","params":{"enabled":false}}}]
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] <== [{"id":"1","error":"NoSuchMethodError: The getter 'isClosed' was called on null.\nReceiver: null\nTried calling: isClosed","trace":"#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:48:5)\n#1      FlutterDevice.views.<anonymous closure> (package:flutter_tools/src/resident_runner.dart:94:46)\n#2      WhereIterator.moveNext (dart:_internal/iterable.dart:439:11)\n#3      ExpandIterator.moveNext (dart:_internal/iterable.dart:477:21)\n#4      new List.from (dart:core/runtime/libarray_patch.dart:40:17)\n#5      Iterable.toList (dart:core/iterable.dart:398:16)\n#6      FlutterDevice.views (package:flutter_tools/src/resident_runner.dart:98:8)\n#7      AppDomain.callServiceExtension (package:flutter_tools/src/commands/daemon.dart:492:61)\n<asynchronous suspension>\n#8      Domain.handleCommand.<anonymous closure> (package:flutter_tools/src/commands/daemon.dart:189:34)\n#9      new Future.sync (dart:async/future.dart:224:31)\n#10     Domain.handleCommand (package:flutter_tools/src/commands/daemon.dart:187:9)\n#11     Daemon._handleRequest (package:flutter_tools/src/commands/daemon.dart:145:26)\n#12     _rootRunUnary (dart:async/zone.dart:1132:38)\n#13     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#14     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)\n#17     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:132:11)\n#18     _MapStream._handleData (dart:async/stream_pipe.dart:232:10)\n#19     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:164:13)\n#20     _rootRunUnary (dart:async/zone.dart:1132:38)\n#21     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#22     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#23     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#24     _BufferingStreamSub…
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] <== [{"id":"2","error":"NoSuchMethodError: The getter 'isClosed' was called on null.\nReceiver: null\nTried calling: isClosed","trace":"#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:48:5)\n#1      FlutterDevice.views.<anonymous closure> (package:flutter_tools/src/resident_runner.dart:94:46)\n#2      WhereIterator.moveNext (dart:_internal/iterable.dart:439:11)\n#3      ExpandIterator.moveNext (dart:_internal/iterable.dart:477:21)\n#4      new List.from (dart:core/runtime/libarray_patch.dart:40:17)\n#5      Iterable.toList (dart:core/iterable.dart:398:16)\n#6      FlutterDevice.views (package:flutter_tools/src/resident_runner.dart:98:8)\n#7      AppDomain.callServiceExtension (package:flutter_tools/src/commands/daemon.dart:492:61)\n<asynchronous suspension>\n#8      Domain.handleCommand.<anonymous closure> (package:flutter_tools/src/commands/daemon.dart:189:34)\n#9      new Future.sync (dart:async/future.dart:224:31)\n#10     Domain.handleCommand (package:flutter_tools/src/commands/daemon.dart:187:9)\n#11     Daemon._handleRequest (package:flutter_tools/src/commands/daemon.dart:145:26)\n#12     _rootRunUnary (dart:async/zone.dart:1132:38)\n#13     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#14     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)\n#17     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:132:11)\n#18     _MapStream._handleData (dart:async/stream_pipe.dart:232:10)\n#19     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:164:13)\n#20     _rootRunUnary (dart:async/zone.dart:1132:38)\n#21     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#22     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#23     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#24     _BufferingStreamSub…
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] <== [{"id":"3","error":"NoSuchMethodError: The getter 'isClosed' was called on null.\nReceiver: null\nTried calling: isClosed","trace":"#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:48:5)\n#1      FlutterDevice.views.<anonymous closure> (package:flutter_tools/src/resident_runner.dart:94:46)\n#2      WhereIterator.moveNext (dart:_internal/iterable.dart:439:11)\n#3      ExpandIterator.moveNext (dart:_internal/iterable.dart:477:21)\n#4      new List.from (dart:core/runtime/libarray_patch.dart:40:17)\n#5      Iterable.toList (dart:core/iterable.dart:398:16)\n#6      FlutterDevice.views (package:flutter_tools/src/resident_runner.dart:98:8)\n#7      AppDomain.callServiceExtension (package:flutter_tools/src/commands/daemon.dart:492:61)\n<asynchronous suspension>\n#8      Domain.handleCommand.<anonymous closure> (package:flutter_tools/src/commands/daemon.dart:189:34)\n#9      new Future.sync (dart:async/future.dart:224:31)\n#10     Domain.handleCommand (package:flutter_tools/src/commands/daemon.dart:187:9)\n#11     Daemon._handleRequest (package:flutter_tools/src/commands/daemon.dart:145:26)\n#12     _rootRunUnary (dart:async/zone.dart:1132:38)\n#13     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#14     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)\n#17     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:132:11)\n#18     _MapStream._handleData (dart:async/stream_pipe.dart:232:10)\n#19     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:164:13)\n#20     _rootRunUnary (dart:async/zone.dart:1132:38)\n#21     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#22     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#23     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#24     _BufferingStreamSub…
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] <== [{"id":"4","error":"NoSuchMethodError: The getter 'isClosed' was called on null.\nReceiver: null\nTried calling: isClosed","trace":"#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:48:5)\n#1      FlutterDevice.views.<anonymous closure> (package:flutter_tools/src/resident_runner.dart:94:46)\n#2      WhereIterator.moveNext (dart:_internal/iterable.dart:439:11)\n#3      ExpandIterator.moveNext (dart:_internal/iterable.dart:477:21)\n#4      new List.from (dart:core/runtime/libarray_patch.dart:40:17)\n#5      Iterable.toList (dart:core/iterable.dart:398:16)\n#6      FlutterDevice.views (package:flutter_tools/src/resident_runner.dart:98:8)\n#7      AppDomain.callServiceExtension (package:flutter_tools/src/commands/daemon.dart:492:61)\n<asynchronous suspension>\n#8      Domain.handleCommand.<anonymous closure> (package:flutter_tools/src/commands/daemon.dart:189:34)\n#9      new Future.sync (dart:async/future.dart:224:31)\n#10     Domain.handleCommand (package:flutter_tools/src/commands/daemon.dart:187:9)\n#11     Daemon._handleRequest (package:flutter_tools/src/commands/daemon.dart:145:26)\n#12     _rootRunUnary (dart:async/zone.dart:1132:38)\n#13     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#14     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)\n#17     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:132:11)\n#18     _MapStream._handleData (dart:async/stream_pipe.dart:232:10)\n#19     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:164:13)\n#20     _rootRunUnary (dart:async/zone.dart:1132:38)\n#21     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#22     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#23     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#24     _BufferingStreamSub…
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] <== [{"id":"5","error":"NoSuchMethodError: The getter 'isClosed' was called on null.\nReceiver: null\nTried calling: isClosed","trace":"#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:48:5)\n#1      FlutterDevice.views.<anonymous closure> (package:flutter_tools/src/resident_runner.dart:94:46)\n#2      WhereIterator.moveNext (dart:_internal/iterable.dart:439:11)\n#3      ExpandIterator.moveNext (dart:_internal/iterable.dart:477:21)\n#4      new List.from (dart:core/runtime/libarray_patch.dart:40:17)\n#5      Iterable.toList (dart:core/iterable.dart:398:16)\n#6      FlutterDevice.views (package:flutter_tools/src/resident_runner.dart:98:8)\n#7      AppDomain.callServiceExtension (package:flutter_tools/src/commands/daemon.dart:492:61)\n<asynchronous suspension>\n#8      Domain.handleCommand.<anonymous closure> (package:flutter_tools/src/commands/daemon.dart:189:34)\n#9      new Future.sync (dart:async/future.dart:224:31)\n#10     Domain.handleCommand (package:flutter_tools/src/commands/daemon.dart:187:9)\n#11     Daemon._handleRequest (package:flutter_tools/src/commands/daemon.dart:145:26)\n#12     _rootRunUnary (dart:async/zone.dart:1132:38)\n#13     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#14     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)\n#17     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:132:11)\n#18     _MapStream._handleData (dart:async/stream_pipe.dart:232:10)\n#19     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:164:13)\n#20     _rootRunUnary (dart:async/zone.dart:1132:38)\n#21     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#22     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#23     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#24     _BufferingStreamSub…
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] <== [{"id":"6","error":"NoSuchMethodError: The getter 'isClosed' was called on null.\nReceiver: null\nTried calling: isClosed","trace":"#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:48:5)\n#1      FlutterDevice.views.<anonymous closure> (package:flutter_tools/src/resident_runner.dart:94:46)\n#2      WhereIterator.moveNext (dart:_internal/iterable.dart:439:11)\n#3      ExpandIterator.moveNext (dart:_internal/iterable.dart:477:21)\n#4      new List.from (dart:core/runtime/libarray_patch.dart:40:17)\n#5      Iterable.toList (dart:core/iterable.dart:398:16)\n#6      FlutterDevice.views (package:flutter_tools/src/resident_runner.dart:98:8)\n#7      AppDomain.callServiceExtension (package:flutter_tools/src/commands/daemon.dart:492:61)\n<asynchronous suspension>\n#8      Domain.handleCommand.<anonymous closure> (package:flutter_tools/src/commands/daemon.dart:189:34)\n#9      new Future.sync (dart:async/future.dart:224:31)\n#10     Domain.handleCommand (package:flutter_tools/src/commands/daemon.dart:187:9)\n#11     Daemon._handleRequest (package:flutter_tools/src/commands/daemon.dart:145:26)\n#12     _rootRunUnary (dart:async/zone.dart:1132:38)\n#13     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#14     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)\n#17     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:132:11)\n#18     _MapStream._handleData (dart:async/stream_pipe.dart:232:10)\n#19     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:164:13)\n#20     _rootRunUnary (dart:async/zone.dart:1132:38)\n#21     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#22     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#23     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#24     _BufferingStreamSub…
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] <== [{"id":"7","error":"NoSuchMethodError: The getter 'isClosed' was called on null.\nReceiver: null\nTried calling: isClosed","trace":"#0      Object.noSuchMethod (dart:core/runtime/libobject_patch.dart:48:5)\n#1      FlutterDevice.views.<anonymous closure> (package:flutter_tools/src/resident_runner.dart:94:46)\n#2      WhereIterator.moveNext (dart:_internal/iterable.dart:439:11)\n#3      ExpandIterator.moveNext (dart:_internal/iterable.dart:477:21)\n#4      new List.from (dart:core/runtime/libarray_patch.dart:40:17)\n#5      Iterable.toList (dart:core/iterable.dart:398:16)\n#6      FlutterDevice.views (package:flutter_tools/src/resident_runner.dart:98:8)\n#7      AppDomain.callServiceExtension (package:flutter_tools/src/commands/daemon.dart:492:61)\n<asynchronous suspension>\n#8      Domain.handleCommand.<anonymous closure> (package:flutter_tools/src/commands/daemon.dart:189:34)\n#9      new Future.sync (dart:async/future.dart:224:31)\n#10     Domain.handleCommand (package:flutter_tools/src/commands/daemon.dart:187:9)\n#11     Daemon._handleRequest (package:flutter_tools/src/commands/daemon.dart:145:26)\n#12     _rootRunUnary (dart:async/zone.dart:1132:38)\n#13     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#14     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#15     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#16     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)\n#17     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:132:11)\n#18     _MapStream._handleData (dart:async/stream_pipe.dart:232:10)\n#19     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:164:13)\n#20     _rootRunUnary (dart:async/zone.dart:1132:38)\n#21     _CustomZone.runUnary (dart:async/zone.dart:1029:19)\n#22     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)\n#23     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)\n#24     _BufferingStreamSub…
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] <== [+1045 ms] [DEVICE LOG] 2018-09-25 14:07:52.746594+0800  localhost Runner[59451]: (AccessibilityUtilities) [com.apple.Accessibility:AXLoading] Will reevaluate code items and load any needed ax code items now
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] <== [        ] [DEVICE LOG] 2018-09-25 14:07:52.747919+0800  localhost Runner[59451]: (AccessibilityUtilities) [com.apple.Accessibility:AXLoading] found 0 axbundle(s) requiring load
[14:07:52 GMT+0800 (CST)] [FlutterRun] [Info] <== [        ] [DEVICE LOG] 2018-09-25 14:07:52.748456+0800  localhost Runner[59451]: (AccessibilityUtilities) [com.apple.Accessibility:AXLoading] Finished loading ax code items
Tue Sep 25 2018 [14:08:50 GMT+0800 (CST)] Log file ended
```

```
Analyzing demo3...
No issues found! (ran in 2.7s)
```

```
[✓] Flutter (Channel beta, v0.8.2, on Mac OS X 10.13.6 17G65, locale en-CN)
    • Flutter version 0.8.2 at /Users/zheng/android/flutter
    • Framework revision 5ab9e70727 (2 weeks ago), 2018-09-07 12:33:05 -0700
    • Engine revision 58a1894a1c
    • Dart version 2.1.0-dev.3.1.flutter-760a9690c2

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.0)
    • Android SDK at /Users/zheng/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.0
    • ANDROID_HOME = /Users/zheng/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[!] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.0, Build version 10A255
    ✗ libimobiledevice and ideviceinstaller are not installed. To install, run:
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    ✗ ios-deploy not installed. To install:
        brew install ios-deploy
    • CocoaPods version 1.6.0.beta.1

[✓] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 28.0.2
    • Dart plugin version 181.5616
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] VS Code (version 1.27.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.18.0

[✓] Connected devices (1 available)
    • iPhone X • 6DBBD95F-4D36-47CF-892C-171BFDE35AB4 • ios • iOS 12.0 (simulator)

! Doctor found issues in 1 category.
```

**Comment by DanTup** [comment]
There is a bug in the current stable VS Code that may result in connecting to Observatory too early when running Flutter in verbose mode. Try installing [this beta](https://github.com/Dart-Code/Dart-Code/releases/tag/v2.19.0-beta.1) version and see if the issue goes away.

**Comment by zsirfs** [comment]
thanks. Let me try.

**Comment by zsirfs** [comment]
Yesterday, I used flscode to develop flutter, because it can't be updated hot, and the application is not updated after restarting, so I can only use vscode to write code, compile and output to iPhoneX emulator with xcode, the process is very troublesome.

**Comment by zsirfs** [comment]
When I use the upgraded dart-code to 2.19 as you said, there is another problem. Just press `f5` and run normally. The log is not reported correctly. However, the simulator has a white screen and can be displayed correctly only on the command line, but it is not yet known whether it can be hot updated.

**Comment by zsirfs** [comment]
I also found another problem, now my flutter will not actively compile my source code, he will directly use the compiled source code, the situation is this: I removed a lot of code in the dart source code, but use `flutter run` Will not take the initiative to update, only after compiling with xcode, will be updated, that is, I still need to click on xcode to compile every time.

**Comment by zsirfs** [comment]
before:
![image](https://user-images.githubusercontent.com/22249411/46053790-d04fea00-c176-11e8-82ac-936c790c0fc9.png)

after:
![image](https://user-images.githubusercontent.com/22249411/46053806-dd6cd900-c176-11e8-9c53-d634e8069c98.png)

**Comment by DanTup** [comment]
@zsirfs You said further up that with the new beta you get stuck on a white screen, but your last screenshot shows the app running. Do you solve this, or did you go back to v2.18?

> I still need to click on xcode to compile every time

Can you provide more details about what you're doing here? What are you doing in Xcode - do you have it open specifically to compile, or something else? Does running `flutter clean` before `flutter run` make any difference?

**Comment by zsirfs** [comment]
I has been running `flutter clean` before `flutter run`, make me my app ran up, not hot update in cli.

If you run the app with vscode, the command will execute normally, but the emulator will display blank after launching the app.

Please see the vscode operation process：

![2018-09-26 19 31 08](https://user-images.githubusercontent.com/22249411/46077401-542fc380-c1c3-11e8-9616-f12518cc37a6.gif)

This is sim:
![image](https://user-images.githubusercontent.com/22249411/46077468-92c57e00-c1c3-11e8-8876-7b7ba93a7892.png)

Please see the cli operation process:

Gif image uploaded later, it is too large.


no hot update in the cli.

**Comment by zsirfs** [comment]
@DanTup Today I switched the branch to dev and tried the vscode again, but I didn't expect it to take effect. Thank you for your help, I hope that you can use these features on the official version, thank you. Thank you for your hard work.

**Comment by zsirfs** [comment]
This issue is waiting for the next beta version to be closed later, thank you for your help.

**Comment by zsirfs** [comment]
@DanTup Yes, I am using your recommended dart-code v2.19.0-beta1. I found that I can't hot-update components like CustomScrollView and need to restart after each change. I did not directly modify the main function, I am in the build, or will report an error. Very strange。

Thank you for your positive response, I will give you any feedback in time. Thank you for your hard work.

**Comment by zsirfs** [comment]
I just found out that one problem in my project is that there is no comment, no component is identified, and there is no code hint.
![image](https://user-images.githubusercontent.com/22249411/46253012-787def80-c4a4-11e8-8cd9-6165794b1353.png)

**Comment by DanTup** [comment]
> I found that I can't hot-update components like CustomScrollView and need to restart after each change

Changes to Stateful widgets will require a "Hot Restart" instead of a "Hot Reload" (it's still very fast, but it'll lose state). I don't think CustomScrollWidget is stateful, but it may depend what you have in it/around it.

> I just found out that one problem in my project is that there is no comment, no component is identified, and there is no code hint.

I'm not sure what you mean by this - can you clarify (or open a new issue in Dart-Code if it's not related to hot-reload failing, since it sounds more of an analysis/editor issue).

**Comment by DanTup** [comment]
@zsirfs Please let me know the status of this - is the `isClosed` crash fixed? (The Dart plugin has been updated yesterday and should auto-update for you, so you're no longer on a beta version).

Is the hot reload issue possibly explained by the use of state (state is persisted so you may need to `Hot Restart` when changing stateful widgets - see https://flutter.io/hot-reload/#previous-state-is-combined-with-new-code)?

As for the final issue, which sounds like you're describing that the editor is not working correctly (no tooltips, no closing labels?) should be raised against [Dart-Code](https://github.com/Dart-Code/Dart-Code).

**Comment by zsirfs** [comment]
@DanTup Now my dev channel is normal here. Thank you for your help.

**Comment by zsirfs** [comment]
I will continue to pay attention to it, I will give you any feedback in time.
